core/transaction.go

```diff
@@ -673,6 +665,28 @@ func transactionCommitment(transactions []Transaction, protocolVersion string) (
 	})
 }
 
+func TransactionCommitmentPoseidon(transactions []Transaction) (*felt.Felt, error) {
```

Comment:
Does this need to be exported? It's tested indirectly via block hash tests right?